test/unit/analysis/allocate_net_gen_test.py

```diff
@@ -332,16 +333,99 @@ def test_missing_energy_source():
         boiler_generator_assn_eia860=boiler_generator_assn_eia860,
     )
 
-    gf, bf, _, gens, _ = allocate_net_gen.extract_input_tables(mock_pudl_out)
+
+def get_ratio_from_bf_and_allocated_by_boiler(
+    bf: pd.DataFrame,
+    allocated: pd.DataFrame,
+    bga: pd.DataFrame,
+    boiler_id_to_check: str,
+    energy_source_code_to_check: str,
+) -> tuple[float, float]:
+    """Helper function to calculate the ratio of a boiler's fuel consumption."""
+    # what gen is this boiler associated with? needed for masking in the allocated tbl
+    generator_id_to_check = bga.loc[
+        (bga.boiler_id == boiler_id_to_check), "generator_id"
+    ]
+
+    def sum_of_fuel_consumed_mmbtu_by_esc(
+        df: pd.DataFrame, energy_source_code_to_check: str
+    ) -> float:
+        return df[
+            (df.energy_source_code == energy_source_code_to_check)
+        ].fuel_consumed_mmbtu.sum()
+
+    ratio_bf = bf[
+        (bf.energy_source_code == energy_source_code_to_check)
+        & (bf.boiler_id == boiler_id_to_check)
+    ].fuel_consumed_mmbtu.sum() / sum_of_fuel_consumed_mmbtu_by_esc(
+        bf, energy_source_code_to_check
+    )
+    ratio_allocated = allocated.loc[
+        (allocated.energy_source_code == energy_source_code_to_check)
+        & allocated.generator_id.isin(generator_id_to_check)
+    ].fuel_consumed_mmbtu.sum() / sum_of_fuel_consumed_mmbtu_by_esc(
+        allocated, energy_source_code_to_check
+    )
+    return ratio_bf, ratio_allocated
+
+
+def test_missing_energy_source(example_2_pudl_tabl):
+    gf, bf, bga, gens, _ = allocate_net_gen.extract_input_tables(example_2_pudl_tabl)
     gens = allocate_net_gen.add_missing_energy_source_codes_to_gens(gens, gf, bf)
     # assert that the missing energy source code is RC
     assert gens.energy_source_code_8.unique() == "RC"
 
     allocated = allocate_net_gen.allocate_gen_fuel_by_generator_energy_source(
-        mock_pudl_out
+        example_2_pudl_tabl
     )
 
     assert (
-        generation_fuel_eia923.fuel_consumed_mmbtu.sum()
+        example_2_pudl_tabl.gf_eia923().fuel_consumed_mmbtu.sum()
         == allocated.fuel_consumed_mmbtu.sum()
     )
+
+    # check if the ratio of the fuel in the gf table for these two generotors is the same
+    # in the gf and the allocated tables
+    ratio_bf, ratio_allocated = get_ratio_from_bf_and_allocated_by_boiler(
+        bf, allocated, bga, boiler_id_to_check="1", energy_source_code_to_check="DFO"
+    )
+    assert ratio_bf == ratio_allocated
+
+
+def test_missing_pm_code_in_bf(example_2_pudl_tabl):
```

Comment:
Similar comment as above - this looks like two separate tests as well. Because it's documenting some weird behavior, I'd probably keep both.